src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,788 @@
+
+TRANSITION_ID_IDLE = "to-idle"
+TRANSITION_ID_RUNNING = "to-running"
+TRANSITION_ID_TRIGGER = "to-trigger"
+TRANSITION_ID_CHECKING_DWELL = "to-checking-dwell"
+TRANSITION_ID_STOPPED = "to-stopped"
+TRANSITION_ID_ERROR = "to-error"
+
+TRANSITION_NAME_IDLE = "To Idle"
+TRANSITION_NAME_RUNNING = "To Running"
+TRANSITION_NAME_TRIGGER = "To Trigger"
+TRANSITION_NAME_CHECKING_DWELL = "To Checking Dwell"
+TRANSITION_NAME_STOPPED = "To Stopped"
+TRANSITION_NAME_ERROR = "To Error"
+
+
+class MightyEbic(Detector):
+    def __init__(self, name: str, role: str, device: str, port: int,
+                 channel: int, url: str, namespace: str, **kwargs):
+        """
+        Initialise the EBIC controller
+        :param name:
+        :param role:
+        :param device:
+        :param port:
+        :param channel:
+        :param url:
+        :param namespace:
+        """
+        super().__init__(name, role, **kwargs)
+        # for debug, eventually -> get resolution from the controller/detector
+        max_res = (1024, 1024)
+        min_res = (1, 1)
+
+        self.device = device
+        self.port = port
+        self.acq_data = None
+        self.idx = None
+        self.sub_handler = SubHandler()
+        self.delay = 5e-8  # need to make this a VA on the stream panel as well?
+        self.t_opc_connection = None
+        self.fake_img = None
+        self.StartScanOPCUA = None
+        self._name = name
+        self._channel = channel
+        self._namespace = namespace
+        self._client = Client(url=url)
+        self._opc_server = None
+        self._binning = (1, 1)
+        self._error_msg = None
+        self._simserver = None
+        self._connected = False
+        self._scan_time = 0.0
+        self._dwell_time = 0.0
+
+        # fake image setup
+        if self.device == "fake":
+            image = "simsem-fake-output.h5"
+            # ensure relative path is from this file
+            if not os.path.isabs(image):
+                image = os.path.join(os.path.dirname(__file__), image)
+            converter = dataio.find_fittest_converter(image, mode=os.O_RDONLY)
+            # just for simulation of the data stream
+            new_img = img.ensure2DImage(converter.read_data(image)[0])  # is now (2048, 2048) -> restrict to res
+            new_img = numpy.array(new_img)
+            self.fake_img = img.rescale_hq(new_img, max_res)
+
+        # The shape is just one point, the depth
+        idt = numpy.iinfo(self.fake_img.dtype)
+        data_depth = idt.max - idt.min + 1
+        self._shape = (data_depth,)  # only one point
+
+        # 8 or 16 bits image
+        if data_depth == 255:
+            bpp = 8
+        else:
+            bpp = 16
+        self.bpp = model.IntEnumerated(bpp, {8, 16})
+        self.scanState = model.StringVA("Disconnected")
+
+        try:
+            self._start_opcclient_thread()
+        except ConnectionError:
+            raise ConnectionError(self._error_msg)
```

Comment:
You need to raise `HwError` if there is a problem in the connection which indicates that the device is not reachable. This way, the backend will show a nice warning message to the user, and try again later.